internal/config/git.go

```diff
@@ -10,13 +10,15 @@ import (
 
 // Push registers Git push functionality for the given repo and key.
 func (cfg *Config) Push(repo string, pk ssh.PublicKey) {
-	err := cfg.Reload()
-	if err != nil {
-		log.Printf("error reloading after push: %s", err)
-	}
-	if cfg.Cfg.Callbacks != nil {
-		cfg.Cfg.Callbacks.Push(repo)
-	}
+	go func() {
```

Comment:
We probably want to put a lock on `cfg.Reload()` since this can happen from multiple threads (it was a problem before, but is probably even more likely to happen in a goroutine.

Reply:
Good point!